### PR TITLE
Remove Debug Assertions from CallSite causing VM crash in Debug mode

### PR DESCRIPTION
**Pull request by reddyashish**
# Purpose

This PR is to fix the crash that was reported in this task:
 https://jira.autodesk.com/browse/DYN-2055

The debug assertions are removed and the output is correct after that. 

### Declarations

Check these if you believe they are true

- [ ] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@aparajit-pratap @mjkkirschner @QilongTang

### CODE REVIEW

File: src/Engine/ProtoCore/Lang/CallSite.cs [hunk 1]
```diff
@@ -8,13 +15,6 @@
 using System.Runtime.Serialization;
 using System.Runtime.Serialization.Formatters.Soap;
 using System.Text;
-using ProtoCore.DSASM;
-using ProtoCore.Exceptions;
-using ProtoCore.Lang;
-using ProtoCore.Lang.Replication;
-using ProtoCore.Properties;
-using ProtoCore.Runtime;
-using ProtoCore.Utils;
```
Comment (aparajit-pratap) [inline src/Engine/ProtoCore/Lang/CallSite.cs]: I think the System `using`s should be placed before ours, @QilongTang ?

Comment (mjkkirschner) [reply src/Engine/ProtoCore/Lang/CallSite.cs]: I sense a code standards fight coming.

Comment (QilongTang) [reply src/Engine/ProtoCore/Lang/CallSite.cs]: @mjkkirschner @aparajit-pratap I think a while ago I have made it settled, I am sorry I probably at some point told @reddyashish the wrong info... Please refer to https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards for the only source of truth.

File: src/Engine/ProtoCore/Lang/CallSite.cs [hunk 2]
```diff
@@ -174,11 +174,6 @@ public ISerializable GetLeftMostData()
 
                 if (!HasNestedData)
                     return null;
-#if DEBUG
-
-                Validity.Assert(NestedData != null, "Nested data has changed null status since last check, suspected race");
-                Validity.Assert(NestedData.Count > 0, "Empty subnested array, please file repo data with @lukechurch, relates to MAGN-4059");
```
Comment (aparajit-pratap) [inline src/Engine/ProtoCore/Lang/CallSite.cs]: @reddyashish I think it makes sense to keep the `NestedData != null` assertion and just remove the second one as there is a check for `NestedData.Count` to be zero in the next line.

File: src/Engine/ProtoCore/Lang/CallSite.cs (hunk 1 above)
Comment (reddyashish) [reply src/Engine/ProtoCore/Lang/CallSite.cs]: @aparajit-pratap Changed it to place the System using's first

**Comment by aparajit-pratap** [comment]
LGTM

**Comment by reddyashish** [comment]
Merging this, @mjkkirschner this is not going in 2.4 right?
Task: https://jira.autodesk.com/browse/DYN-2055

**Pull request merged**